test/end-to-end/incident-query-filter.test.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+var utils = require('./e2e-tools');
+var request = require('supertest');
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+var expect = chai.expect;
+var promise = protractor.promise;
+
+// Allow chai to wait for promises on the right-hand-side
+chaiAsPromised.transformAsserterArgs = function(args) {
+  return promise.all(args);
+};
+
+describe.only('Incident query filter', function() {
+  before(utils.initDb);
+  after(utils.disconnectDropDb);
+
+  beforeEach(utils.resetDb);
+  beforeEach(utils.initAdmin.bind({}, 'asdfasdf'));
+  beforeEach(utils.toggleFetching.bind({}, 'Off'));
+  beforeEach(utils.addSource.bind({}, 'SMS GH', { nickname: 'foo', keywords: 'test' }));
+  beforeEach(utils.toggleFetching.bind({}, 'On'));
+
+  afterEach(utils.deleteSource.bind({}, 'SMS GH', 'foo'));
+  afterEach(utils.toggleFetching.bind({}, 'Off'));
+  afterEach(utils.resetBrowser);
+
+  var reqParams1 = {
+    from: '9845098450',
+    fulltext: 'loremipsumdolor',
+    date: '2016-09-01',
+    keyword: 'test'
+  };
+
+  var reqParams2 = {
+    from: '12345678',
+    fulltext: 'foobarbaz',
+    date: '2016-09-02',
+    keyword: 'test'
+  };
+
+  var reqParams3 = {
+    from: '987654321',
+    fulltext: 'world',
+    date: '2016-09-03',
+    keyword: 'test'
+  };
+
+  var reqParams4 = {
+    from: '0864213579',
+    fulltext: 'whatwhat',
+    date: '2016-09-04',
+    keyword: 'test'
+  };
+
+  var incident1 = {
+    title: 'hello',
+    tags: 'hello, there, how, do, you, doo',
+    location: 'macau'
+  };
+
+  var incident2 = {
+    title: 'response',
+    tags: 'hello, I, am, good, thank, you',
+    location: 'hong kong'
+  };
```

Comment:
I think there are enough things here to put these in a separate `test/end-to-end/fixtures/incidents.json` as in the backend tests.